src/data/int/basic.lean

```diff
@@ -351,6 +351,10 @@ begin
   cases hk; exact ⟨_, hk⟩
 end
 
+lemma eq_of_nat_abs_eq_nonneg {a b : ℤ} (ha : 0 ≤ a) (hb : 0 ≤ b) (h : nat_abs a = nat_abs b) :
+  a = b :=
+by rwa [←sq_eq_sq ha hb, ←nat_abs_eq_iff_sq_eq]
```

Comment:
Why not state it as an `iff`?
```suggestion
lemma nat_abs_eq_nat_abs_iff_of_nonneg {a b : ℤ} (ha : 0 ≤ a) (hb : 0 ≤ b) :
  nat_abs a = nat_abs b ↔ a = b :=
by rw [←sq_eq_sq ha hb, ←nat_abs_eq_iff_sq_eq]
```

Reply:
I've added the other three variants.  Not very elegant proofs, I'm sure there's a better way to do it that I haven't seen.